frontend/src/components/DragAndDrop/DragAndDrop.js

```diff
@@ -0,0 +1,196 @@
+import { useState, useRef, useEffect } from "react";
+import { CloudArrowUpFill } from "react-bootstrap-icons";
+import { useSpring, animated } from "react-spring";
+import "./DragAndDrop.css";
+import NavBar from "../NavBar";
+import App from "../../App";
+
+function DragAndDrop() {
+  const [showToast, setShowToast] = useState(false);
+  const [dragActive, setDragActive] = useState(false);
+  const [fileUploaded, setFileUploaded] = useState(false);
+  const [invalidJson, setInvalidJson] = useState(false);
+  const [lines, setLines] = useState();
+  const inputRef = useRef(null);
+
+  const flyingAnimation = useSpring({
+    transform: fileUploaded ? "translateY(-100%)" : "translateY(0%)",
+  });
+
+  const handleShowToast = () => {
+    setShowToast(true);
+    setTimeout(() => {
+      setShowToast(false);
+    }, 4000);
+  };
+
+  const handleDragEnter = function (e) {
+    e.preventDefault();
+    e.stopPropagation();
+    setDragActive(true);
+  };
+
+  const handleDragOver = function (e) {
+    e.preventDefault();
+    e.stopPropagation();
+    setDragActive(true);
+  };
+
+  const handleDragLeave = function (e) {
+    e.preventDefault();
+    e.stopPropagation();
+    setDragActive(false);
+  };
+
+  const handleDrop = function (e) {
+    e.preventDefault();
+    e.stopPropagation();
+    setDragActive(false);
+    if (e.dataTransfer.files && e.dataTransfer.files[0]) {
+      const file = e.dataTransfer.files[0];
+      handleFiles(file);
+      setFileUploaded(true);
+    }
+  };
+
+  const handleChange = function (e) {
+    e.preventDefault();
+    e.stopPropagation();
+    if (e.target.files && e.target.files[0]) {
+      const file = e.target.files[0];
+      handleFiles(file);
+      setFileUploaded(true);
+    }
+  };
+
+  const onButtonClick = () => {
+    inputRef.current.value = "";
+    inputRef.current.click();
+  };
+
+  const handleFiles = (file) => {
+    const fileName = file.name;
+    if (fileName.endsWith(".json") || fileName.endsWith(".jsonl")) {
+      const reader = new FileReader();
+      reader.onload = (e) => {
+        try {
+          const dataObjectsArray = e.target.result.trim().split(/\r?\n/);
+          setLines(dataObjectsArray.map((line) => JSON.parse(line)));
+        } catch (error) {
+          handleShowToast();
```

Comment:
I don't understand why there are 2 representations of errors? Here you use toast, but if uploaded file has wrong extension then it looks completely different. Can we use just one approach for representing errors?

Reply:
`not have enough info` has not been handled yet. Though it's having very less chance of occurrence.

Reply:
I haven't tried to preview what this looks like yet, so I could be off base, but definitely don't be checking the extension of the file -- you don't know what it will be, someone can redirect into a file of any name.

And I don't think it's very different if the file isn't valid line-delimited JSON or if it's somehow otherwise invalid -- these files are internal implementation details of Bowtie (i.e. the contents of them aren't really hand authored by anyone, Bowtie produces them and Bowtie then consumes them) -- so an invalid file basically only has one explanation, namely "you're uploading something that wasn't produced by Bowtie" -- if anything else is happening it's likely a Bowtie bug.

Reply:
> I haven't tried to preview what this looks like yet, so I could be off base, but definitely don't be checking the extension of the file -- you don't know what it will be, someone can redirect into a file of any name (I regularly redirect into some file called output with no extension).

Hmm, I didn't think that way.

So to check the validity of the report, I think we must be checking if it contains at least three json lines (one containing `started` property, one containing `case` property and one containing `implementation` property). And if any of these is missing then we can consider it as invalid.

Reply:
Right, don't check any of that explicitly though -- simply start parsing the file assuming it is valid -- if you get an error, the file is invalid (this is essentially what the Python code does) and then you report to the user "hey you gave me something other than a Bowtie report".